### DISCUSSION TITLE
Close CustomToast component from external controller

### DISCUSSION THREAD

**Discussion by RichK67** [Blazored/Toast#218] [General]
We have a controller that causes a number of toasts to be generated showing error messages. We want, when the user corrects the input field and the toast is still showing, that the individual toast is closed.

We think the only way we can do this is to pass the ToastInstance, and then it is a doddle to call ToastInstance.Close().
However, ToastInstance is declared as an internal class, and therefore cannot be used for any meaningful access.

Alternatively, if we could access the ToastContainer, we could also find the ToastInstance, and clear the correct individual toast. Or perhaps there be a way to discover the ToastInstance GUID, and send a command to toastService to close the toast with that GUID.

Is there any way we can acheive this?
We would rather not have to pull our own copy of the repo, modifying the code, and rebuilding.

All help appreciated.
Richard

**Comment by artemfomin** [comment]
Agree. For custom toasts it'd be useful in general to have an option to close it. if you could add `ToastId` as cascading parameter and add method `CloseToast(Guid id)` to `IToastService` it would at least give an opportunity to extend your component without changing your code or using reflection as I did

**Reply by artemfomin** [reply]
To be more clear I had a case to close the toast by right clicking on it. However inside custom toast element there was completely no handle for it. I can contribute if that's acceptable

**Comment by frankvoelker** [comment]
Any news on this?
